internal/trie/node/node.go

```diff
@@ -43,7 +43,7 @@ type Node struct {
 
 // Type returns Leaf or Branch depending on what type
 // the node is.
-func (n *Node) Type() Type {
+func (n *Node) Type() Kind {
```

Comment:
I like this

Reply:
Oh wait, `Type()` should be `Kind()` now 😄 

It was just deepsource complaining that `type` is a built in name ugh